src/rrules/avoiding_non_differentiable_code.jl

```diff
@@ -14,25 +14,190 @@ end
 @zero_adjoint MinimalCtx Tuple{Type{Float16},Any,RoundingMode}
 @zero_adjoint MinimalCtx Tuple{typeof(==),Type,Type}
 
+# logging, String related primitive rules
+using Base: getindex, getproperty
+using Base.Threads: Atomic
+using Mooncake: zero_fcodual, MinimalCtx, @is_primitive, NoPullback, CoDual
+using Base.CoreLogging: LogLevel, handle_message, invokelatest
+import Base.CoreLogging as CoreLogging
+
+# Rule for accessing an Atomic{T} wrapped Integer with Base.getindex as deriving a rule results
+# in encountering a Atomic->Int address bitcast followed by a llvm atomic load call 
+@is_primitive MinimalCtx Tuple{typeof(getindex),Atomic{T}} where {T<:Integer}
+function rrule!!(::CoDual{typeof(getindex)}, x::CoDual{Atomic{T}}) where {T<:Integer}
+    pb!!(ȳ::NoRData) = NoRData(), NoRData()
+    return zero_fcodual(getindex(x.x)), pb!!
+end
+
+@is_primitive MinimalCtx Tuple{typeof(Base.normpath),String}
+function rrule!!(::CoDual{typeof(Base.normpath)}, path::CoDual{String})
+    pb!!(ȳ::NoRData) = NoRData(), NoRData()
```

Comment:
Is there a reason not to use `NoPullback()`?

Reply:
or maybe look into https://chalk-lab.github.io/Mooncake.jl/stable/utilities/defining_rules/#Functions-with-Zero-Adjoint?